packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/mapper/ProxyMethodMapper.java

```diff
@@ -68,397 +57,136 @@ public static ProxyMethodMapper getInstance() {
-            return createBinaryContentAsyncReturnType();
-        } else if (SchemaUtil.responseContainsHeaderSchemas(operation, settings)) {
-            // SchemaResponse
-            // method with schema in headers would require a ClientResponse
-            if (settings.isGenericResponseTypes()) {
-                // If the response body type is InputStream it needs to be converted to Flux<ByteBuffer> to be
-                // asynchronous, unless this is sync-stack.
-                if (responseBodyType == ClassType.INPUT_STREAM) {
-                    responseBodyType = GenericType.FLUX_BYTE_BUFFER;
-                }
-                IType genericResponseType = GenericType.RestResponse(
-                    Mappers.getSchemaMapper().map(ClientMapper.parseHeader(operation, settings)), responseBodyType);
-
-                if (ignoreTypedHeaders || settings.isDisableTypedHeadersMethods()) {
-                    if (responseBodyType == GenericType.FLUX_BYTE_BUFFER) {
-                        return createStreamContentAsyncReturnType();
-                    }
-                    genericResponseType = GenericType.Response(responseBodyType);
-                }
-                return createSingleValueAsyncReturnType(genericResponseType);
-            } else {
-                ClassType clientResponseClassType = ClientMapper.getClientResponseClassType(operation,
-                    ClientModels.getInstance().getModels(), settings);
-                return createClientResponseAsyncReturnType(clientResponseClassType);
-            }
+    private void buildExpectedResponseFields(Operation operation, JavaSettings settings, ProxyMethod.Builder builder) {
+        final boolean isDataPlaneClient = settings.isDataPlaneClient();
+        final IType bodyType = MapperUtils.getExpectedResponseBodyType(operation, settings);
+        final IType bodyTypeMapped;
+        if (isDataPlaneClient && settings.isBranded()) {
+            builder.rawResponseBodyType(bodyType);
+            // branded (azure flavor) uses BinaryData as return type not the model.
+            bodyTypeMapped = SchemaUtil.tryMapToBinaryData(bodyType, operation);
         } else {
-            if ((!settings.isDataPlaneClient()
-                && !settings.isSyncStackEnabled()
-                && settings.isInputStreamForBinary()
-                && responseBodyType.equals(ClassType.BINARY_DATA)) || responseBodyType.equals(ClassType.INPUT_STREAM)) {
-                return createStreamContentAsyncReturnType();
-            } else if (responseBodyType.equals(PrimitiveType.VOID)) {
-                IType singleValueType = GenericType.Response(ClassType.VOID);
-                return createSingleValueAsyncReturnType(singleValueType);
-            } else {
-                IType singleValueType = GenericType.Response(responseBodyType);
-                return createSingleValueAsyncReturnType(singleValueType);
-            }
+            // unbranded uses the actual model as return type.
```

Comment:
nit, mgmt/vanilla/unbranded all use the actual model

Reply:
thanks, updated, being explicit is always good :)